ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -96,32 +104,84 @@ const onNavigationStateChange = (
 class IdpLoginScreen extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
-
     this.state = {
-      requestState: pot.noneLoading
+      requestState: pot.noneLoading,
+      startingMs: new Date().getTime()
     };
   }
 
-  private handleOnError = (): void =>
+  private getSpidId = (): string =>
+    this.props.loggedOutWithIdpAuth
+      ? this.props.loggedOutWithIdpAuth.idp.entityID
+      : "idpID n/a";
```

Comment:
since this is a magic string, use uppercase with underscore (easier to spot and use in queries)

```suggestion
      : "UNDEFINED_IDP";
```